azure-pipelines.yml

```diff
@@ -1,213 +1,347 @@
+                  echo $URI
+                  echo "##vso[task.setvariable variable=STAGING_URI;isOutput=true]$URI"
+            - task: AzureWebAppContainer@1
+              displayName: Deploy to staging slot
+              inputs:
+                azureSubscription: $(DEV_AZURE_SUBSCRIPTION)
+                deployToSlotOrASE: true
+                resourceGroupName: $(DEV_RESOURCE_GROUP_NAME)
+                appName: $(ANALYZER_DEV_APP_NAME)
+                slotName: staging
+                imageName: $(REGISTRY_NAME)presidio-analyzer:$(Build.BuildId)
+
+        - job: DeployAnonymizerToDev
+          displayName: Deploy Anonymizer to Dev Environment
+          pool:
+            vmImage: 'ubuntu-16.04'
+          steps:
+            - checkout: none
+            - task: AzureCLI@2
+              displayName: Create staging slot
+              name: createStagingSlot
+              inputs:
+                azureSubscription: $(DEV_AZURE_SUBSCRIPTION)
+                scriptType: bash
+                scriptLocation: inlineScript
+                inlineScript: |
+                  set -eux  # fail on error
+
+                  az webapp deployment slot create --name $(ANONYMIZER_DEV_APP_NAME) --resource-group $(DEV_RESOURCE_GROUP_NAME) --configuration-source $(ANONYMIZER_DEV_APP_NAME) --slot staging
+                  URI=https://$(az webapp config hostname list --webapp-name $(ANONYMIZER_DEV_APP_NAME) --resource-group $(DEV_RESOURCE_GROUP_NAME) --slot staging --query '[0].name' -o tsv)
+                  echo $URI
+                  echo "##vso[task.setvariable variable=STAGING_URI;isOutput=true]$URI"
+            - task: AzureWebAppContainer@1
+              displayName: Deploy to staging slot
+              inputs:
+                azureSubscription: $(DEV_AZURE_SUBSCRIPTION)
+                deployToSlotOrASE: true
+                resourceGroupName: $(DEV_RESOURCE_GROUP_NAME)
+                appName: $(ANONYMIZER_DEV_APP_NAME)
+                slotName: staging
+                imageName: $(REGISTRY_NAME)presidio-anonymizer:$(Build.BuildId)
+        
+        - job: WaitForDeployment
+          displayName: Wait for deployment
+          dependsOn: 
+            - DeployAnalyzerToDev
+            - DeployAnonymizerToDev
+          pool: Server
+          steps:
+          - task: Delay@1
+            inputs:
+              delayForMinutes: '5'
+
+        - job: TestDeployment
+          displayName: Smoke and E2E Tests on the deployement
+          dependsOn: 
+            - DeployAnalyzerToDev
+            - DeployAnonymizerToDev
+            - WaitForDeployment
+          variables:
+            ANALYZER_STAGING_URI: $[ dependencies.DeployAnalyzerToDev.outputs['createStagingSlot.STAGING_URI'] ]  
+            ANONYMIZER_STAGING_URI: $[ dependencies.DeployAnonymizerToDev.outputs['createStagingSlot.STAGING_URI'] ]  
+          steps:
+          - task: Bash@3
+            displayName: 'Smoke test'
```

Comment:
Can this depend on external file, because if any changes to the interface occurs, you need to change the CI!

Reply:
great comment! i removed the specific requests and switched to using /health endpoint of both services. (had to add it to anonymizer as well).